app/models/git_repository.rb

```diff
@@ -25,6 +25,20 @@ def self.delete_repo_dir(id, directory_name)
     FileUtils.rm_rf(directory_name)
   end
 
+  # Check the connection for the remote.
+  #
+  # Does not clone a repo or use an existing one, and instead initializes a
+  # repo in a temp path to check the remote to see if it can accept
+  # connections.
+  def check_connection
+    Dir.mktmpdir do |tmp_repo_dir|
```

Comment:
All of this `mktmpdir` stuff should now be able to be removed, as this was only necessary to create initialize a git repo so `rugged`/`git_worktree.rb` could initialize a dummy repo properly and use `Rugged::Remote#check_conection` (which requires a `Rugged::Repository` instance... unfortunately.

See this conversation in the previous PR for context as to "why":  https://github.com/ManageIQ/manageiq/pull/20645#discussion_r499000187